SDBlockDevice.cpp

```diff
@@ -558,6 +558,17 @@ int SDBlockDevice::erase(bd_addr_t addr, bd_size_t size)
     }
     int status = BD_ERROR_OK;
 
+    // Align erases, missed blocks will be left unerased, but this is fine as
+    // the block device api doesn't garuntee the final state of erased bytes and
+    // a following program will still work fine.
+    bd_size_t misalignment = _erase_size - (addr % _erase_size);
```

Comment:
@geky Agreed with @LemonBoy. SD controller will take care of alignment and deciding what range of erase blocks should be erased in particular request.

We can at most throw an error, if size and address are not aligned to erase size, which should be done by is_valid_erase() check.